python/test/pipeline_test.py

```diff
@@ -805,50 +807,46 @@ def testAbortRootAsyncAlreadyAborted(self):
 
   def testFinalizeEmailDone_HighReplication(self):
     """Tests completion emails for completed root pipelines on HRD."""
-    old_app_id = os.environ['APPLICATION_ID']
-    testutil.TEST_APP_ID = 's~my-hrd-app'
-    os.environ['APPLICATION_ID'] = testutil.TEST_APP_ID
-    testutil.setup_for_testing(define_queues=['other'])
-    try:
-      stage = OutputlessPipeline()
-      stage.start(idempotence_key='banana')
-      stage._context.transition_complete(stage._pipeline_key)
-      other = OutputlessPipeline.from_id(stage.pipeline_id)
 
-      result = []
-      def fake_mail(self, sender, subject, body, html=None):
+    self.test_app_id = 's~my-hrd-app'
```

Comment:
I am not sure why this test really needed to change the app id (maybe it does not and instead validation should be done against the regular app id defined in testutil.
If you think it is needed you can seperate the method into its own test class and override the TEST_APP_ID constant (as I showed in the example).